### PR TITLE
Add styles and mixins

### PR DESCRIPTION
**Pull request by rowanmanning**
This adds in the styles for basic and heavy stepped progress indicators. I think this is enough for us to release a beta. This doesn't contain documentation and Sass tests, I'd like to PR those separately as this is already quite a large PR.

Also IE9 isn't tested and (based on Lee's feedback) is broken. I'd like to PR this work separately as well.

## Master styles

Master styles

## Internal Styles

Internal styles

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -11,5 +11,10 @@
 	"main": [
 		"main.scss",
 		"main.js"
-	]
+	],
+	"dependencies": {
+		"o-colors": "^4.7.7",
+		"o-icons": "^5.8.0",
+		"o-typography": "^5.7.8"
```
Comment (notlee) [inline bower.json:18]: Could we include `o-normalise@^1.0.0"` too, for focus states etc?

File: demos/src/demo.scss [hunk 2]
```diff
@@ -2,3 +2,16 @@
...
 
 @import '../../main';
+
+body {
+	@include oColorsFor('page', 'background');
+	margin: 1em;
+}
+
+.demo-container {
+	margin-bottom: 2em;
```
Comment (notlee) [inline demos/src/demo.scss:12]: What are your thoughts on adding `demo-container` styles to `body`? To remove superfluous markup from demos.

Comment (notlee) [reply demos/src/demo.scss:12]: Ohhh... and maybe just keep this for the "testing" demo?

File: src/scss/_brand.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+@function _oSteppedProgressSupports($variant) {
+	@return oBrandSupportsVariant($component: 'o-stepped-progress', $variant: $variant);
+}
+
+// Define shared brand configurations
+$_o-stepped-progress-shared-brand-config: (
+	font-scale: -1,
+	step-size: 22px,
+	step-border-size: 2px,
+	label-max-width: 140px,
```
Comment (notlee) [inline src/scss/_brand.scss]: An `em` unit might be a little better here, to accommodate a [custom typography scale](https://github.com/Financial-Times/o-typography/pull/166).

**Review by notlee — COMMENTED** [review]
Loving it! Just a couple of small suggestions, but also looks like something's gone wrong at some point for IE9?

screenshot 2018-12-10 at 13 53 25

File: demos/src/demo.scss (hunk 2 above)
Comment (rowanmanning) [reply demos/src/demo.scss:12]: These are only used in the pa11y and testing demos already, not in any of the public demos

File: src/scss/_brand.scss (hunk 3 above)
Comment (rowanmanning) [reply src/scss/_brand.scss]: done 👍

**Review by notlee — APPROVED** [review]

**Pull request merged**